wordpress-mobile/WordPress-Android
# Issue/17836 jp install plugin viewed event

Part of #17836

This adds the `jp_install_full_plugin_card_viewed, Properties: {"tab_source":"<source>"}` event 

where <source> can be:
1. "dashboard"
2. "site_menu"
3. "nonexistent": this is the case when the tabbed dashboard is not available (like when we are with self-hosted sites without  the full JP installed)

Tbh I was not able to figure out real scenarios different from the last one for the case of individual plugins. But preferred to keep the general case and tested the other 2 scenarios by:
a. Pointing to the SingleActionCard for the "Please delete the WP app" card
b. Removing the if filtering for JETPACK_INSTALL_FULL_PLUGIN_CARD in JetpackInstallFullPluginShownTracker.trackShown

| Image for reference of the 3rd case | <img width="262" alt="image" src="https://user-images.githubusercontent.com/47797566/219909486-550634ec-4b9b-43cb-993c-091df3452874.png"> |
|---|---|

To test:

1 - Install app and make sure the feature flag JetpackInstallFullPluginFeatureConfig is enabled and so the tracking;
2 - Select a site with only individual plugins: notice the card for install full JP plugin and check in logcat the tracking `jp_install_full_plugin_card_viewed, Properties: {"tab_source":"nonexistent"}`
3 - Select a different non problem site and check no tracking is triggered; move back to the site with only individual plugins and check the tracking is triggered correctly.

## Regression Notes
1. Potential unintended areas of impact
none

5. What I did to test those areas of impact (or what existing automated tests I relied on)
manual testing

6. What automated tests I added (or what prevented me from doing so)
Unit testing will be added in separate PR.

PR submission checklist:
